File: scripts/test_solutions/981_time_based_key_value_store.cpp
class TimeMap {
private:
    unordered_map<string, vector<pair<int, string>>> store;

public:
    TimeMap() {
    }

    void set(string key, string value, int timestamp) {
        store[key].push_back({timestamp, value});
    }

    string get(string key, int timestamp) {
        if (store.find(key) == store.end()) {
            return "";
        }
        auto& arr = store[key];
        int lo = 0;
        int hi = arr.size() - 1;
        string result = "";
        while (lo <= hi) {
            int mid = (lo + hi) / 2;
            if (arr[mid].first <= timestamp) {
                result = arr[mid].second;
                lo = mid + 1;
            } else {
                hi = mid - 1;
            }
        }
        return result;
    }
};

File: scripts/test_solutions/981_time_based_key_value_store.py
from collections import defaultdict

class TimeMap:
    def __init__(self):
        self.store = defaultdict(list)

    def set(self, key, value, timestamp):
        self.store[key].append((timestamp, value))

    def get(self, key, timestamp):
        if key not in self.store:
            return ''
        arr = self.store[key]
        lo, hi = 0, len(arr) - 1
        result = ''
        while lo <= hi:
            mid = (lo + hi) // 2
            if arr[mid][0] <= timestamp:
                result = arr[mid][1]
                lo = mid + 1
            else:
                hi = mid - 1
        return result
